### PR TITLE
fix: update report header to new design

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

Update rule header under failed instances section to new design.
The main issue we're solving is the collapse/expand button is not right before the content it collapse/expand, i.e. after activating the button to expand the hidden content, the user (keyboard or screen reader) needs to navigate to the rule link, rule description and WCAG links before reaching the instance list, at which point context may be lost (i.e. the user may already forgot the hidden content was expanded).

Main strategy: 
- simplify the header so it doesn't contains any interactive element (links basically)
- make the whole header a button so it's easier to activate by mouse users
- update styles to make the control discoverable (mouse cursor, hover state, focus state signal the element can be activated/clickable

Notes:
- Figma mocks show a different focus state style, this change has been defer for now
- Adding the card that show the rule and WCAG links will be part of a different PR

**overall, collapsed**
01 - overall, collapsed

**overall, expanded**
02 - overall, expanded

**focus state**
03 - selected state

**hover state**
04 - hover state

**simple keyboard navigation**
05 - keyboard navigation

**keyboard navigation, expand/collapse**
06 - keyboard, collapse expand

**passed/not applicable section**
07 - bottom section

#### Pull request checklist

- [x] Addresses an existing issue: Part of WI # 1558756
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [x] (UI changes only) Added screenshots/GIFs to description above
- [x] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/DetailsView/reports/components/report-sections/full-rule-detail.tsx [hunk 1]
```diff
@@ -13,17 +13,17 @@ import { InstanceOutcomeType } from '../instance-outcome-type';
...
-export type RuleDetailProps = {
-    deps: RuleDetailDeps;
+export type FullRuleDetailProps = {
+    deps: FullRuleDetailDeps;
     rule: RuleResult;
     outcomeType: InstanceOutcomeType;
 };
 
-export const RuleDetail = NamedSFC<RuleDetailProps>('RuleDetails', props => {
+export const FullRuleDetail = NamedSFC<FullRuleDetailProps>('RuleDetails', props => {
```
Comment (dbjorge) [inline src/DetailsView/reports/components/report-sections/full-rule-detail.tsx]: The name of the NamedSFC is still `'RuleDetails'`

Comment (smoralesd) [reply src/DetailsView/reports/components/report-sections/full-rule-detail.tsx]: :( I'll look at all the `NamedSFC` I updated/created on this PR to make sure I got them all correctly.

**Comment by dbjorge** [comment]
LGTM besides the one NamedSFC thing!

File: src/DetailsView/reports/components/report-sections/full-rule-detail.tsx (hunk 1 above)
Comment (smoralesd) [reply src/DetailsView/reports/components/report-sections/full-rule-detail.tsx]: Updated!

**Pull request merged**